hack/test_e2e.sh

```diff
@@ -116,7 +116,7 @@ function setup_ginkgo {
 }
 
 function apply_ipvx_fixes {
-    if ci_mode = true ; then
+    if "${ci_mode}" = true ; then
```

Comment:
@FriedrichWilken what's the deal with `ci_mode` here?

Reply:
@knabben We use this check to know this function should run in github ci or not

Reply:
Why can't it be made common across local and CI?
I see 3 usages of this variable
1. it is installing kind, kubectl etc if ci_mode is true. We can just check if binary not present, then install.
2. ip forwarding settings - just echo saying we are setting it to true for the test (if you think this shouldn't be modified on the host, then how about something like preserving what the initial value was and then restoring it after the test?)
3. clean artifacts - I'm not quite sure what cleanup it is doing. But need not be separate for CI and local IMO.

Reply:
For item 2. I agree that we need to restore. For item 3. Not sure either, currently, it's exporting logs :)
By the way, I would deprecate this PR in favor of: https://github.com/kubernetes-sigs/kpng/pull/163

Reply:
I will just drop this patch in favor of 163